docs/users/cross-compilation.md

```diff
@@ -0,0 +1,144 @@
+---
+id: cross-compilation
+title: Cross compilation setups
+---
+
+Cross-building the same codebase with Scala 2.13 and Scala 3.3+ is common, but
```

Comment:
I have a personal preference for cross-building rather than cross-compilation as it puts less emphasis on the compile phase - let's just make sure we use the same term in the header and the body